enclave/src/integration-test/java/org/hyperledger/besu/enclave/TlsEnabledEnclaveTest.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.enclave;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.hyperledger.besu.enclave.TlsHelpers.populateFingerprintFile;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Optional;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TlsEnabledEnclaveTest {
+
+  private TlsEnabledHttpServerFactory serverFactory;
+  private Vertx vertx;
+
+  final TlsCertificateDefinition orionCert =
+      TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
+  final TlsCertificateDefinition besuCert =
+      TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
+
+  public void shutdown() {
+    vertx.close();
+  }
+
+  @Before
+  public void setup() {
+    serverFactory = new TlsEnabledHttpServerFactory();
+    this.vertx = Vertx.vertx();
+  }
+
+  @After
+  public void cleanup() {
+    serverFactory.shutdown();
+    this.shutdown();
+  }
+
+  private Enclave createEnclave(final int orionPort, final Path workDir, final boolean tlsEnabled)
+      throws IOException {
+
+    final Path serverFingerprintFile = workDir.resolve("server_known_clients");
+    final Path besuCertPasswordFile = workDir.resolve("password_file");
+    try {
+      populateFingerprintFile(serverFingerprintFile, orionCert, Optional.of(orionPort));
+      Files.write(besuCertPasswordFile, besuCert.getPassword().getBytes(Charset.defaultCharset()));
+
+      final EnclaveFactory factory = new EnclaveFactory(vertx);
+      if (tlsEnabled) {
+        final URI orionUri = new URI("https://localhost:" + orionPort);
+        return factory.createVertxEnclave(
+            orionUri,
+            besuCert.getPkcs12File().toPath(),
+            besuCertPasswordFile,
+            serverFingerprintFile);
+      } else {
+        return factory.createVertxEnclave(new URI("http://localhost:" + orionPort));
+      }
+    } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException e) {
+      fail("unable to populate fingerprint file");
+      return null;
+    } catch (URISyntaxException e) {
+      fail("unable to create URI");
+      return null;
+    }
+  }
+
+  @Test
+  public void nonTlsEnclaveCannotConnectToTlsServer() throws IOException {
+
+    Path workDir = Files.createTempDirectory("test-certs");
+
+    // Note: the HttpServer always responds with a JsonRpcSuccess, result="I'm up".
+    final HttpServer orionHttpServer = serverFactory.create(orionCert, besuCert, workDir, true);
```

Comment:
rename to enclaveServer or maybe just httpServer here and in the other tests?

Reply:
done